ckanext/unaids/react/components/BulkFileUploaderComponent/src/App.js

```diff
@@ -40,38 +40,68 @@ export default function App(props) {
-        // } else if (fuzzySearchResults.length > 1) {
-        //     const firstResult = fuzzySearchResults[0];
-        //     const secondResult = fuzzySearchResults[1];
-        //     const scroreDiff = firstResult.score - secondResult.score;
-        //     if (scroreDiff > 0.05) {
-        //         defaultUploadAction = fuzzySearchResults[0].item;
-        //     }
-        // }
-        // return defaultUploadAction;
-        const exactMatches = updateResourcesOptions.filter(
-            x => x.fileName === file.name
-        );
-        switch (exactMatches.length) {
-            case 1:
-                return exactMatches[0];
-            default:
-                return extraResource;
+
+        function strDiff(str1, str2){
+            // Get all chars that match at the start of both strings
+            var openingMatch = "";
+            for(let x = 0; x < str1.length; x++){
+                if(str1.charAt(x) == str2.charAt(x)) openingMatch += str1.charAt(x);
+                else break;
+            }
+
+            // Get all chars that match at the end of both strings
+            var str1Reversed = str1.split("").reverse().join("");
+            var str2Reversed = str2.split("").reverse().join("");
+            var endingMatch = "";
+            for(let x = 0; x < str1.length; x++){
+                if(str1Reversed.charAt(x) == str2Reversed.charAt(x)) endingMatch += str1Reversed.charAt(x);
+                else break;
+            }
+            endingMatch = endingMatch.split("").reverse().join("");
+
+            // Get the middle non-matching part of each string
+            var str1Difference = str1.slice(openingMatch.length, str1.length-endingMatch.length).trim();
+            var str2Difference = str2.slice(openingMatch.length, str2.length-endingMatch.length)
+
+            return [openingMatch, endingMatch, str1Difference, str2Difference];
+        }
+
+        function filenameMatch(filename1, filename2){
+            /* Matches where filenames begin and end with the same three (or more chars),
+               have the same file extension, and any middle non-matching parts
+               consist only of the following inserted or swapped:
+               - a number
+               - a number or the word "copy" in parenthesis
+               - a number or the word "copy" after a hyphen
+            */
+
+            const regexp = /^$|^ ?[\(-]? ?([0-9]|copy|Copy)*\)?$/;
+            const matches = strDiff(filename1, filename2);
+            const completeMatch = filename1 == filename2;
+            const openingMatch = matches[0].length >= 3;
+            const closingMatch = matches[1].length >= 3;
+            const extensionMatch = matches[1].includes('.');
+            const regexMatch = matches[2].match(regexp) != null && matches[3].match(regexp) != null;
+            const match = completeMatch || openingMatch && closingMatch && extensionMatch && regexMatch;
```

Comment:
As a personal note, I always find it confusing chaining so many operators together because you can never be too sure what's being compared with what 😅  So i'd either add a bunch of brackets, declare more variables or do something like this:

````
const match = [
    completeMatch || openingMatch,
    closingMatch,
    extensionMatch,
    regexMatch
].includes(false) === false
```

You may need to run this to confirm the `||` works as expected though

Reply:
AND always before OR, but brackets for clarity is a good idea